test/end-to-end/company.js

```diff
@@ -1933,4 +1933,294 @@ describe("Company endpoint", () => {
+                    .send({
+                        name: "Changing Company",
+                        bio: "Without permission",
+                        contacts: ["1"],
+                        logo: "http://awebsite.com/otherlogo.jpg",
+                    })
+                    .expect(HTTPStatus.FORBIDDEN);
+
+                expect(res.body.errors[0]).toHaveProperty("msg", ValidationReasons.INSUFFICIENT_PERMISSIONS_COMPANY_SETTINGS);
+            });
+
+            test("Should fail if not logged in", async () => {
+                const res = await test_agent
+                    .put(`/company/${test_company._id}/edit`)
+                    .send({
+                        name: "Changing Company",
+                        bio: "Without login",
+                        contacts: ["1"],
+                        logo: "http://awebsite.com/otherlogo.jpg",
+                    })
+                    .expect(HTTPStatus.UNAUTHORIZED);
+
+                expect(res.body.errors[0]).toHaveProperty("msg", ValidationReasons.INSUFFICIENT_PERMISSIONS);
+            });
+        });
+
+        describe("Should pass if good user", () => {
+            test("Should pass if god", async () => {
+                await test_agent
+                    .put(`/company/${test_company._id}/edit`)
+                    .send(withGodToken({
+                        name: "Changing Company",
+                        bio: "As god",
+                        contacts: ["1"],
+                        logo: "http://awebsite.com/otherlogo.jpg",
+                    }))
+                    .expect(HTTPStatus.OK);
+            });
+
+            test("Should pass if same company", async () => {
+                await test_agent
+                    .post("/auth/login")
+                    .send(test_user_company)
+                    .expect(HTTPStatus.OK);
+
+                const res = await test_agent
+                    .put(`/company/${test_company._id}/edit`)
+                    .send({
+                        name: "Changing Company",
+                        bio: "As company itself",
+                        logo: "http://awebsite.com/otherlogo.jpg",
+                    })
+                    .expect(HTTPStatus.OK);
+
+                expect(res.body).toHaveProperty("name", "Changing Company");
+                expect(res.body).toHaveProperty("bio", "As company itself");
+                expect(res.body).toHaveProperty("logo", "http://awebsite.com/otherlogo.jpg");
+            });
+        });
+
+        test("Offer should be updated", async () => {
+            await test_agent
+                .post("/auth/login")
+                .send(test_user_company)
+                .expect(HTTPStatus.OK);
+
+            const res = await test_agent
+                .put(`/company/${test_company._id}/edit`)
+                .send({
+                    name: "Changing Company",
+                    contacts: ["122"],
```

Comment:
Just a suggestion, but you could maybe factor these values out into another variable, since they are reused later in the test.

Reply:
Did a map of 'changing_values' and used that.